src/main/java/com/ldtteam/structurize/placement/handlers/placement/PlacementHandlers.java

```diff
@@ -714,6 +714,27 @@ public static void handleTileEntityPlacement(
         }
     }
 
+    /**
+     * Get the required items for this state.
+     * @param world the world it will be placed in.
+     * @param pos the pos to place it at.
+     * @param state the state to place.
+     * @param data its TE data.
+     * @param complete if complete.
+     * @return the required items.
+     */
+    public static List<ItemStack> getRequiredItemsForState(final World world, final BlockPos pos, final BlockState state, final CompoundNBT data, final boolean complete)
+    {
+        for (final IPlacementHandler placementHandler : PlacementHandlers.handlers)
+        {
+            if (placementHandler.canHandle(world, pos, state))
+            {
+                return placementHandler.getRequiredItems(world, pos, state, data, complete);
```

Comment:
This can cause an infinite loop as you are calling this from getRequiredItems

Reply:
Theoretically yes, practically this won't happen.

Reply:
Funny not funny, go merge otherwise